### PR TITLE
Add callback for unsubscribe from newsletter via SendGrid

### PR DESCRIPTION
**Pull request by dasyad00**
Webhook attached with signature validation from SendGrid

## Description
Added a new endpoint for SendGrid webhook. Signature saved in KeePass DB.

## Motivation and Context
SendGrid now updates the website database in the event that the user decides to unsubscribe from the newsletter.

## How Has This Been Tested?
WIth help from ngrok (documentation added in `webhooks.py`

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dasyad00** [comment]
OAuth authentication is yet to be done.

References:
https://sendgrid.com/docs/for-developers/tracking-events/getting-started-event-webhook-security-features/#oauth-client-credentials-flow
https://github.com/jazzband/django-oauth-toolkit

**Review by bcvandendool — APPROVED** [review]

File: apps/users/urls.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-from . import views
+from . import views, webhooks
 
 app_name = "cosmos_users"
 
 urlpatterns = [
     path("register/", views.register, name="user_register"),
     path("profile/", views.process_profile_form, name="user_profile"),
     path("delete/", views.delete, name="user_delete"),
+    url(r"hook/$", csrf_exempt(webhooks.SendGridWebhook.as_view()), name="email_hook2"),
```
Comment (bcvandendool) [inline apps/users/urls.py:13]: Why email_hook2?

File: cosmos/urls.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 urlpatterns = [
     path("admin/", admin.site.urls),
+    path("o/", include("oauth2_provider.urls", namespace="oauth2_provider")),
```
Comment (bcvandendool) [inline cosmos/urls.py:17]: maybe make this url a bit more descriptive? makes it easier to debug if later there are any issues with it

**Review by bcvandendool — APPROVED** [review]

File: apps/users/urls.py (hunk 1 above)
Comment (dasyad00) [reply apps/users/urls.py:13]: I was testing the difference between with and without OAuth. I'll fix it up in the next commit.

**Pull request merged**